tests/contrib/celery/test_integration.py

```diff
@@ -0,0 +1,225 @@
+from nose.tools import eq_, ok_
+
+from .utils import CeleryTestCase
+
+
+class CeleryIntegrationTask(CeleryTestCase):
+    """
+    Ensures that the tracer works properly with a real Celery application
+    without breaking the Application or Task APIs.
+    """
+    def test_concurrent_delays(self):
+        # it should create one trace for each delayed execution
+        @self.app.task
+        def fn_task():
+            return 42
+
+        for x in range(100):
+            fn_task.delay()
+
+        traces = self.tracer.writer.pop_traces()
```

Comment:
Actually tasks are not executed, unless we're running a real worker / using the Celery threaded workers (available only on Celery 4.0+). All `delay()` calls return immediately and tasks are enqueued for later processing.

Because of that, we generally have 2 spans for each execution (one for the enqueuing and one for the execution), but this is the expected behavior.